src/spatial_reaction_network.jl

```diff
@@ -0,0 +1,109 @@
+# Currently only permit constant rate.
+struct SpatialReaction
+    rate::Symbol
+    substrates::Tuple{Vector{Pair{Symbol,Int64}},Vector{Pair{Symbol,Int64}}}
+    products::Tuple{Vector{Pair{Symbol,Int64}},Vector{Pair{Symbol,Int64}}}
+    net_stoich_src::Vector{Pair{Symbol, Int64}}
+    net_stoich_dst::Vector{Pair{Symbol, Int64}}
+
+    function SpatialReaction(rate::Symbol, substrates_in::Tuple{Vector,Vector}, products_in::Tuple{Vector,Vector})
+        substrates = process_sr_species.(substrates_in)
+        products = process_sr_species.(products_in)
+        new(rate, substrates, products, net_stoich(substrates,products,1), net_stoich(substrates,products,2))
+    end
+end
+process_sr_species(species::Vector) = map(s -> (s isa Symbol) ? s => 1 : s, species);
+function net_stoich(substrates, products, idx)
+    sub_dict = Dict(Pair.(first.(substrates[idx]), -last.(substrates[idx])))
+    prod_dict = Dict(products[idx])
+    return collect(mergewith(-, prod_dict, sub_dict))
+end;
+
+function diffusion_reaction(rate, species)
+    return SpatialReaction(rate, ([species], []), ([]), [species])
+end;
+
+### Lattice Reaction Network Structure ###
+# Couples:
+# A reaction network (that is simulated within each compartment).
+# A set of spatial reactions (denoting interaction between comaprtments).
+# A network of compartments (a meta graph that can contain some additional infro for each compartment).
+# The lattice is a DiGraph, normals graphs are converted to DiGraphs (with one edge in each direction).
+struct LatticeReactionSystem # <: MT.AbstractTimeDependentSystem # Adding this part messes up show, disabling me from creating LRSs
+    """The spatial reactions defined between individual nodes."""
+    rs::ReactionSystem
+    """The spatial reactions defined between individual nodes."""
+    spatial_reactions::Vector{SpatialReaction}
+    """The graph on which the lattice is defined."""
+    lattice::DiGraph
+    """Dependent (state) variables representing amount of each species. Must not contain the
+    independent variable."""
+    
+    function LatticeReactionSystem(rs, spatial_reactions, lattice::DiGraph)
+        return new(rs, spatial_reactions, lattice)
+    end
+    function LatticeReactionSystem(rs, spatial_reactions, lattice::SimpleGraph)
+        return new(rs, spatial_reactions, DiGraph(lattice))
+    end
+end
+
+### ODEProblem ###
+# Creates an ODEProblem from a LatticeReactionSystem.
+function DiffEqBase.ODEProblem(lrs::LatticeReactionSystem, u0, tspan,
+                               p = DiffEqBase.NullParameters(), args...;
+                               kwargs...)
+    @unpack rs, spatial_reactions, lattice = lrs       
+                        
+    spatial_params = unique(getfield.(spatial_reactions, :rate))
+    pV_in, pE_in = split_parameters(p, spatial_params)
+    nV, nE = length.([vertices(lattice), edges(lattice)])
+    u_idxs = Dict(reverse.(enumerate(Symbol.(getfield.(states(rs), :f)))))

```

Comment:
Use `Symbolics.getname` instead of accessing `:f` to use the established API function.

Reply:
Right